cpp/oneapi/dal/array.hpp

```diff
@@ -24,6 +24,10 @@
 namespace oneapi::dal {
 namespace v1 {
 
+/// @tparam T The type of the memory block elements within the array.
+///              $T$ can represent any type.
```

Comment:
`$...$` is used for mathematical expressions, to refer to the template parameter or normal parameters of the function, use ``:literal:`...` ``. It should be changed in the spec as well